implementations/rust/ockam/ockam_identity/src/secure_channel/handshake/handshake_worker.rs

```diff
@@ -131,10 +131,13 @@ impl Worker for HandshakeWorker {
         Ok(())
     }
 
-    async fn shutdown(&mut self, _context: &mut Self::Context) -> Result<()> {
+    async fn shutdown(&mut self, context: &mut Self::Context) -> Result<()> {
         if let Some(handler) = &self.decryptor_handler {
-            handler.shutdown().await?
+            handler.shutdown().await?;
+            self.secure_channels.secure_channel_registry.unregister_channel(&handler.addresses.encryptor);
```

Comment:
Let's move this and the next call to the beginning of the function. We want to call those functions even if handler.shutdown returned an error

Reply:
Cool, just one more question.
If we want to ignore errors what is the preferred way?
1. `context.stop_worker(self.addresses.encryptor.clone()).await.err();`
2. `let _ = context.stop_worker(self.addresses.encryptor.clone()).await;`
@SanjoDeundiak

Reply:
```
let _ = context.stop_worker(self.addresses.encryptor.clone()).await;
```
This one, tnx for asking.
Just to clarify: my comment was not about ignoring the error from `handler.shutdown().await?;`, but to making sure that even if an error occurred in that call - we still run the other 2 actions